mendix/widgets-tools
# [WTF-2074] Rollup 3 Upgrade

## Checklist

-   Contains unit tests ✅ ❌
-   Contains breaking changes ✅ 
-   Compatible with: MX 🔟 
-   Did you update version and changelog? ✅ 
-   PR title properly formatted (`[XX-000]: description`)? ✅ 

## This PR contains


## What is the purpose of this PR?

This PR aims to upgrade the bundler setup in order to offer more recent packages and extend the longevity of the package.

## Relevant changes

Package versions have been updated and modifications have been made to the Rollup and jest configurations to be compatible with newer versions.

## What should be covered while testing?

How to test the branch locally:

1. Pull repository and checkout this branch `wtf/rollup-3`
2. Run `npm pack`
3. Copy the full path of the resulting archive (The previous command states the name of the file)
4. In the widget directory, run `npm install $full_path_to_archive`

Properties to check:

- Newly generated widgets should build and test without any modification necessary. 
- Widgets without custom configuration should build and test without any modification necessary.
- Widgets with custom configuration (for example [carousel-web](https://github.com/mendix/web-widgets/blob/main/packages/pluggableWidgets/carousel-web/rollup.config.js)) should be upgrade-able using the guide in CHANGELOG.md 

## Extra comments (optional)
